src/config/exec.c

```diff
@@ -31,95 +32,98 @@ cfgExecParam(ConfigCommand commandId, ConfigCommandRole commandRoleId, const Key
+                if (optionReplace != NULL)
+                {
+                    exists = kvKeyExists(optionReplace, key);
 
-            // If the key exists but is NULL then skip this option
-            if (exists && value == NULL)
-                continue;
+                    if (exists)
+                        value = kvGet(optionReplace, key);
+                }
 
-            // If no replacement then see if this option is valid for the current command and is not default
-            if (value == NULL && cfgOptionValid(optionId))
-            {
-                if (cfgOptionNegate(optionId))
-                    value = BOOL_FALSE_VAR;
-                else if (cfgOptionSource(optionId) != cfgSourceDefault)
-                    value = cfgOption(optionId);
-            }
+                // If the key exists but is NULL then skip this option
+                if (exists && value == NULL)
+                    continue;
 
-            // If the option was reset
-            if (cfgOptionReset(optionId))
-            {
-                strLstAdd(result, strNewFmt("--reset-%s", cfgOptionName(optionId)));
-            }
-            // Else format the value if found
-            else if (value != NULL && (!local || exists || cfgOptionSource(optionId) == cfgSourceParam))
-            {
-                if (varType(value) == varTypeBool)
+                // If no replacement then see if this option is not default
+                if (value == NULL && cfgOptionValid(optionId))
                 {
-                    strLstAdd(result, strNewFmt("--%s%s", varBool(value) ? "" : "no-", cfgOptionName(optionId)));
+                    if (cfgOptionIdxNegate(optionId, optionIdx))
+                        value = BOOL_FALSE_VAR;
+                    else if (cfgOptionIdxSource(optionId, optionIdx) != cfgSourceDefault)
+                        value = cfgOptionIdx(optionId, optionIdx);
                 }
-                else
-                {
-                    StringList *valueList = NULL;
-
-                    if (varType(value) == varTypeKeyValue)
-                    {
-                        valueList = strLstNew();
-
-                        const KeyValue *optionKv = varKv(value);
-                        const VariantList *keyList = kvKeyList(optionKv);
 
-                        for (unsigned int keyIdx = 0; keyIdx < varLstSize(keyList); keyIdx++)
-                        {
-                            strLstAdd(
-                                valueList,
-                                strNewFmt(
-                                    "%s=%s", strZ(varStr(varLstGet(keyList, keyIdx))),
-                                    strZ(varStrForce(kvGet(optionKv, varLstGet(keyList, keyIdx))))));
-                        }
-                    }
-                    else if (varType(value) == varTypeVariantList)
+                // If the option was reset
+                if (cfgOptionValid(optionId) && cfgOptionIdxReset(optionId, optionIdx))
```

Comment:
Because they may not be valid for the prior command, but they are valid for the new command and the option was explicitly set in `optionReplace`.

The reason why this was added is in the old code cfgOptionReset() would return false even if the option was not valid, but now it will segfault because there is no memory allocated for the index list. So, really, this was a bug in the old code that was never exposed because cfgOptionReset() was very permissive.

Reply:
I clarified the comments in a separate commit so you can revert if needed: 66ad2922996d68b1933a7f67d062499a7c85d5db

Reply:
Looks good to me.